packages/core/src/internal/fetchSource.ts

```diff
@@ -57,21 +25,41 @@ export const makeFetchSource = (
           fetchOptions.signal = abortController.signal;
         }
 
-        return executeFetch(operation, url, fetchOptions);
+        return fetcher(url, fetchOptions).then((res: Response) => {
+          statusNotOk =
+            res.status < 200 ||
+            res.status >= (fetchOptions.redirect === 'manual' ? 400 : 300);
+
+          return (response = res);
+        });
       })
-      .then((result: OperationResult | undefined) => {
-        if (!ended) {
-          ended = true;
-          if (result) next(result);
-          complete();
-        }
-      });
+      .then(async (res: Response | undefined) => {
```

Comment:
I'd avoid mixing `async/await` and `then` chaining in the same blocks in a file. We do have a Babel plugin that converts these to regular `then` chains, but I'd prefer to make due without `async/await` where possible or alternatively check transpilation output. I think at least in this case we can definitely avoid this

Reply:
its just a tad tricky to handle that "early return" in this scenario without it. `.then` chains will continue to execute. If the response is ended, we need to "break out" of the .then chain. Think the only way is to "throw".